kolena/workflow/test_case.py

```diff
@@ -332,3 +493,48 @@ def edit(self, reset: bool = False) -> Iterator[Editor]:
         test_case_data = from_dict(data_class=CoreAPI.EntityData, data=complete_res.json())
         self._populate_from_other(self._create_from_data(test_case_data))
         log.success(f"edited test case '{self.name}' (v{self.version})")
+
+    @classmethod
+    @contextmanager
+    def bulk_process(cls) -> Iterator[None]:
+        """
+        Create, load or edit multiple test cases in a context:
+
+        ```python
+        # create or load multiple test cases
+        with TestCase.bulk_process():
+            test_case_1 = TestCase("test-case 1", ...)
+            test_case_2 = TestCase("test-case 2", ...)
+
+        # edit multiple test cases
+        with TestCase.bulk_process():
+            for test_case in test_cases:
+                with test_case.edit() as editor:
+                    # perform as many editing actions as desired
+                    editor.add(...)
+                    editor.remove(...)
+        ```
+
+        Changes are committed to the Kolena platform collectively when the context is exited. If there is an error,
+        none of the edits would take effect.
+        """
+        if not hasattr(cls, "workflow"):
+            raise NotImplementedError(f"{cls.__name__} must implement class attribute 'workflow'")
+
+        bulk_processor = _BulkProcessor()
+        token = _BULK_PROCESSOR.set(bulk_processor)
+        try:
+            yield
+
+            log.info(f"loading {len(bulk_processor.data)} test cases")
+
+            bulk_process_response = bulk_processor.process(cls.workflow.name)
+            bulk_processor.commit(bulk_process_response.test_cases)
+
+            log.success(f"edited {len(bulk_processor.data)} test cases")
```

Comment:
When playing around with this locally, this value was zero:

```
kolena> loading 10 test cases
kolena> edited 0 test cases
```

Is that expected?

Reply:
Thanks for catching, fixed and added test.